lightning/src/chain/channelmonitor.rs

```diff
@@ -2736,20 +2736,23 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 	fn is_resolving_htlc_output<L: Deref>(&mut self, tx: &Transaction, height: u32, logger: &L) where L::Target: Logger {
 		'outer_loop: for input in &tx.input {
 			let mut payment_data = None;
-			let revocation_sig_claim = (input.witness.len() == 3 && HTLCType::scriptlen_to_htlctype(input.witness[2].len()) == Some(HTLCType::OfferedHTLC) && input.witness[1].len() == 33)
-				|| (input.witness.len() == 3 && HTLCType::scriptlen_to_htlctype(input.witness[2].len()) == Some(HTLCType::AcceptedHTLC) && input.witness[1].len() == 33);
-			let accepted_preimage_claim = input.witness.len() == 5 && HTLCType::scriptlen_to_htlctype(input.witness[4].len()) == Some(HTLCType::AcceptedHTLC);
+			let witness_items = input.witness.len();
+			let last_witness_len = input.witness.last().map(|w| w.len()).unwrap_or(0);
+			let prev_last_witness_len = input.witness.second_to_last().map(|w| w.len()).unwrap_or(0);
+			let revocation_sig_claim = (witness_items == 3 && HTLCType::scriptlen_to_htlctype(last_witness_len) == Some(HTLCType::OfferedHTLC) && prev_last_witness_len == 33)
```

Comment:
Just as a remark: these lines are pretty long. But I see that wrapping them would likely even worsen readability, so probably best to leave them as is.

Reply:
further DRYed this area, which shortened the lines